### ISSUE TITLE
ON CLUSTER ddl don't pass the settings

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#6107]
It makes it impossible to create tables with experimental features:

```
SET allow_experimental_data_skipping_indices = 1

Ok.

0 rows in set. Elapsed: 0.001 sec. 

chi-test-replicated-0-0-0.chi-test-replicated-0-0.default.svc.cluster.local :) create table test ON CLUSTER replicated(number UInt64, INDEX mm_number (number) TYPE minmax GRANULARITY 1) ENGINE = MergeTree ORDER BY number;

CREATE TABLE test ON CLUSTER replicated
(
    `number` UInt64, 
    INDEX mm_number number TYPE minmax GRANULARITY 1
)
ENGINE = MergeTree
ORDER BY number

┌─host────────────────────┬─port─┬─status─┬─error────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─num_hosts_remaining─┬─num_hosts_active─┐
│ chi-test-replicated-0-0 │ 9000 │     36 │ Code: 36, e.displayText() = DB::Exception: You must set the setting `allow_experimental_data_skipping_indices` to 1 before using data skipping indices. (version 19.11.2.7 (official build)) │                   3 │                0 │
│ chi-test-replicated-1-0 │ 9000 │     36 │ Code: 36, e.displayText() = DB::Exception: You must set the setting `allow_experimental_data_skipping_indices` to 1 before using data skipping indices. (version 19.11.2.7 (official build)) │                   2 │                0 │
│ chi-test-replicated-1-1 │ 9000 │     36 │ Code: 36, e.displayText() = DB::Exception: You must set the setting `allow_experimental_data_skipping_indices` to 1 before using data skipping indices. (version 19.11.2.7 (official build)) │                   1 │                0 │
│ chi-test-replicated-0-1 │ 9000 │     36 │ Code: 36, e.displayText() = DB::Exception: You must set the setting `allow_experimental_data_skipping_indices` to 1 before using data skipping indices. (version 19.11.2.7 (official build)) │                   0 │                0 │
└─────────────────────────┴──────┴────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────────┴──────────────────┘
↘ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.)  0%Received exception from server (version 19.11.2):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: There was an error on [chi-test-replicated-0-0:9000]: Code: 36, e.displayText() = DB::Exception: You must set the setting `allow_experimental_data_skipping_indices` to 1 before using data skipping indices. (version 19.11.2.7 (official build)). 

4 rows in set. Elapsed: 0.196 sec. 
```

**Comment by filimonov** [comment]
Workaround:
add a line `<allow_experimental_data_skipping_indices>1</allow_experimental_data_skipping_indices>` into users.xml in profiles / default subsection.

**Comment by slimtom95** [comment]
To complement this issue:
My ON CLUSTER DLL statement ends up timeout as configured in the default profile. **But change and reload default profile cannot overwrite this max_execution_time setting.** Restart CH servers may work, cannot test this now.
19.14.7.15

**Comment by Felixoid** [comment]
> add a line `<allow_experimental_data_skipping_indices>1</allow_experimental_data_skipping_indices>` into users.xml in profiles / default subsection.

Just a little clarification: the DDDL are executed as the default user, so it's necessary to add the setting to whatever profile related to the `default`

**Comment by tavplubix** [comment]
Fixed in  #21535, set `distributed_ddl_entry_format_version` to 2 and it will pass settings.
